ppmclibs/tinycv_impl.cc

```diff
@@ -649,11 +649,11 @@ static uint16_t read_u16(const unsigned char* data, size_t& offset,
 {
     uint16_t pixel;
     if (do_endian_conversion) {
-        pixel = data[offset++] * 256;
-        pixel += data[offset++];
+        pixel =  (data[offset++] >> 0);
+        pixel |= (data[offset++] >> 8);
```

Comment:
Shifting an unsigned char by 8 will always result in 0. I guess you mean `<< 8`?

Reply:
which would be * 256 (the compiler does the << 8 for you if you compile with any level of O